integration/spark/shared/src/main/java/io/openlineage/spark/agent/util/PathUtils.java

```diff
@@ -79,7 +79,8 @@ public static DatasetIdentifier fromCatalogTable(
     } else {
       return di.withSymlink(
           nameFromTableIdentifier(catalogTable.identifier()),
-          StringUtils.substringBeforeLast(di.getName(), File.separator),
+          StringUtils.substringBeforeLast(
+              catalogTable.storage().locationUri().get().toString(), File.separator),
```

Comment:
Please make sure execution chain does not cause null pointer exception, especially for `locationUri()` which returns an Optional.